apiserver/facades/agent/uniter/register.go

```diff
@@ -26,20 +26,24 @@ func Register(registry facade.FacadeRegistry) {
 }
 
 // newUniterAPI creates a new instance of the core Uniter API.
-func newUniterAPI(_ context.Context, ctx facade.ModelContext) (*UniterAPI, error) {
-	m, err := ctx.State().Model()
+func newUniterAPI(stdCtx context.Context, ctx facade.ModelContext) (*UniterAPI, error) {
+	serviceFactory := ctx.ServiceFactory()
+	modelInfoService := serviceFactory.ModelInfo()
+	modelInfo, err := modelInfoService.GetModelInfo(stdCtx)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	serviceFactory := ctx.ServiceFactory()
 	secretBackendService := serviceFactory.SecretBackend()
 	secretBackendAdminConfigGetter := func(stdCtx context.Context) (*provider.ModelBackendConfigInfo, error) {
-		return secretBackendService.GetSecretBackendConfigForAdmin(stdCtx, model.UUID(m.UUID()))
+		return secretBackendService.GetSecretBackendConfigForAdmin(stdCtx, modelInfo.UUID)
```

Comment:
@wallyworld & @ycliuhw can we have a look at fixing this please? This seems like it can be provided off of the model service factory.

Reply:
@barrettj12 in the meantime can we make this closure actually do the call to GetModelInfo inside instead of outside. Late bind as much as we can.

Reply:
It's not a domain service - it's a method on a service to provide necessary data to pass to the unit agent to use. Are you really saying you want to add it to the model service factory?